dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -289,9 +289,26 @@ public static void publicAccessibleUrls(WorkflowVersion existingVersion,
      * @param repository Repository path (ex. dockstore/dockstore-ui2)
      * @param gitReference Git reference from GitHub (ex. refs/tags/1.0)
      * @param username Git user who triggered the event
+     * @param installationId GitHub App installation ID
      * @param deliveryId The GitHub delivery ID, used to group all lambda events created during this GitHub webhook delete
      */
-    protected void githubWebhookDelete(String repository, String gitReference, String username, String deliveryId) {
+    protected void githubWebhookDelete(String repository, String gitReference, String username, Long installationId, String deliveryId) {
+        if (installationId != null) {
```

Comment:
In what situation is the installationId null? 
In that situation does the following code make sense either way? 
i.e. if there's no valid github app, does it make sense to process the rest of the deletion anyway?

Reply:
There was no installation ID prior to this PR, and we processed the deletion, which we should do if w can.  There will be no installation ID if the endpoint is invoked manually without it, or if the lambda hasn't been updated to propagate it.

Reply:
Why not just default an installation ID in the case of manual tests or a not-updated lambda?

Reply:
If it's optional, you can force the delete to be processed, no matter what the repo state, by omitting `installationId`, which is probably a good capability to have.  I will document that capability somewhere in the commentz.